mafContrib/updatedDitherStackers.py

```diff
@@ -0,0 +1,1196 @@
+##############################################################################################################
+# Motivation: implement various dithering strategies.
+
+# Updated
+# Better repulsive random algorithm in terms of execution style
+# New season stacker
+
+# Some of the stackers here are modified versions of the stackers in lsst.sims.maf.stackers.ditherStackers.
+# These function similarly to ditherStackers,  with the major difference that dithers here are restricted to
+# within the hexagons used to tile the sky. Each of these hexagons is inscribed in a FOV, and having dithers
+# restricted to within these hexagons allows avoiding oversampling near the boundaries of the hexagonal tiles.
+ 
+# Also, the stackers here follow new naming scheme:  [Pattern]Dither[Field]Per[Timescale]. The absence of the
+# keyword 'Field' implies dither assignment to all fields.
+
+# Humna Awan: humna.awan@rutgers.edu
+# Date last modified: 04/09/16
+###############################################################################################################
+import numpy as np
+import matplotlib.pyplot as plt
+from lsst.sims.maf.stackers import BaseStacker
+from mafContrib.SeasonStacker_v2 import SeasonStacker_v2 as SeasonStacker
+import time
+
+__all__ = ['RandomDitherFieldPerVisitStacker',
+           'RepulsiveRandomDitherFieldPerVisitStacker',
+           'SpiralDitherFieldPerVisitStacker',
+           'SequentialHexDitherFieldPerVisitStacker',
+           'RandomDitherFieldPerNightStacker',
+           'RepulsiveRandomDitherFieldPerNightStacker',
+           'SpiralDitherFieldPerNightStacker',
+           'SequentialHexDitherFieldPerNightStacker',
+           'RandomDitherPerNightStacker',
+           'RepulsiveRandomDitherPerNightStacker',
+           'SpiralDitherPerNightStacker',
+           'SequentialHexDitherPerNightStacker',
+           'PentagonDitherFieldPerSeasonStacker',
+           'PentagonDiamondDitherFieldPerSeasonStacker',
+           'PentagonDitherPerSeasonStacker',
+           'PentagonDiamondDitherPerSeasonStacker',
+           'SpiralDitherPerSeasonStacker',
+           'FermatSpiralDitherFieldPerVisitStacker',
+           'FermatSpiralDitherFieldPerNightStacker',
+           'FermatSpiralDitherPerNightStacker']
```

Comment:
Many of these stackers are now implemented in core maf. They are also present in newDithers.py. Having multiple versions will be confusing to other users (which one is the right one to use?), is harder to document, and could lead to name collisions, including problems with the maf framework.

The RepulsiveRandomDithers, Pentagon*Dithers, and FermatSpiralDithers are not implemented anywhere else, and should be in one file. (perhaps delete the newDithers.py versions or combine that file and the new ones here?). 

Note that the core maf dithers allow the option of making the offsets only be within the inscribed hexagon (true by default), so the RandomDither*, SpiralDither* and SequentialhexDither* should now be the same as the dither patterns implemented in core maf  - see https://github.com/lsst/sims_maf/blob/master/python/lsst/sims/maf/stackers/ditherStackers.py).